include/readdy/common/algorithm.h

```diff
@@ -122,5 +123,53 @@ inline void performEvents(Events &events, const ShouldEvaluate &shouldEvaluate,
     }
 }
 
+template<typename ParticleContainer, typename EvaluateOnParticle, typename InteractionContainer,
+        typename EvaluateOnInteraction, typename TopologyContainer, typename EvaluateOnTopology>
+inline void evaluateOnContainers(ParticleContainer &particleContainer,
```

Comment:
why aren't interaction container and topology container crefs?

Reply:
nevermind!

Reply:
good point, fix incoming

Reply:
Actually this can be done with universal references so that the correct type is deduced:
```cpp
template<typename F, typename Collection>
void doThis(F&& f, Collection&& c) {
    for(auto &&x : c) {
        f(x);
    }
};
```